tests/backup/backup_multiple_provisioner_test.go

```diff
@@ -222,3 +222,407 @@ var _ = Describe("{MultipleProvisionerCsiSnapshotDeleteBackupAndRestore}", func(
+			for _, appSpec := range appSpecListDefaultVsc {
+				taskName := fmt.Sprintf("%s-%s", TaskNamePrefix, RandomString(randomStringLength))
+				appCtx := ScheduleApplicationsWithScheduleOptions(taskName, appSpec, provisioner)
+				appCtx[0].ReadinessTimeout = AppReadinessTimeout
+				scheduledAppContextsForDefaultVscBackup = append(scheduledAppContextsForDefaultVscBackup, appCtx...)
+				allAppContext = append(allAppContext, appCtx...)
+			}
+		}
+		// Deploy multiple application in multiple namespace for custom backup
+		for provisioner, _ := range provisionerSnapshotClassMap {
+			var err error
+			appSpecListCustomVsc, err = GetApplicationSpecForProvisioner(clusterProviderName, provisioner)
+			log.FailOnError(err, fmt.Sprintf("Fetching application spec for provisioner %s", provisioner))
+			for _, appSpec := range appSpecListCustomVsc {
+				taskName := fmt.Sprintf("%s-%s", TaskNamePrefix, RandomString(randomStringLength))
+				appCtx := ScheduleApplicationsWithScheduleOptions(taskName, appSpec, provisioner)
+				appCtx[0].ReadinessTimeout = AppReadinessTimeout
+				scheduledAppContextsForCustomVscBackup = append(scheduledAppContextsForCustomVscBackup, appCtx...)
+				allAppContext = append(allAppContext, appCtx...)
+			}
+		}
+	})
+
+	It("Backup and restore of namespaces with multiple provisioners using csi and kdmp", func() {
+		Step("Validate deployed applications", func() {
+			ValidateApplications(allAppContext)
+		})
+		Step("Creating backup location and cloud setting", func() {
+			log.InfoD("Creating backup location and cloud setting")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, provider := range providers {
+				cloudCredName = fmt.Sprintf("%s-%s-%v", "cred", provider, time.Now().Unix())
+				backupLocationName = fmt.Sprintf("%s-%s-bl-%v", provider, getGlobalBucketName(provider), time.Now().Unix())
+				cloudCredUID = uuid.New()
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = backupLocationName
+				err := CreateCloudCredential(provider, cloudCredName, cloudCredUID, BackupOrgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of cloud credential named [%s] for org [%s] with [%s] as provider", cloudCredName, BackupOrgID, provider))
+				err = CreateBackupLocation(provider, backupLocationName, backupLocationUID, cloudCredName, cloudCredUID, getGlobalBucketName(provider), BackupOrgID, "", true)
+				dash.VerifyFatal(err, nil, "Creating backup location")
+			}
+		})
+		Step("Registering cluster for backup", func() {
+			log.InfoD("Registering cluster for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			err = CreateApplicationClusters(BackupOrgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+
+			clusterStatus, err := Inst().Backup.GetClusterStatus(BackupOrgID, SourceClusterName, ctx)
```

Comment:
Can we move it into a loop?

Reply:
It has multiple variations and complexity and to merge it into a loop

Reply:
Ok